nodebb-plugin-fdk-resource-link/static/lib/main.js

```diff
@@ -0,0 +1,118 @@
+                    return;
+    
+                const type = $('#plugin-fdk-resource-link-editor-type').val();
+                const response = await fetch(`${window.FdkResourceLink.getSearchBaseUrl()}/suggestions/${encodeURIComponent(type)}?q=${encodeURIComponent(e.target.value)}`);
+                const suggestions = await response.json();
+                const items = suggestions?.suggestions.map(({ id, title, searchType }) => {
+                    return `<option data-id="${id}" data-type="${searchType}" value="${window.FdkResourceLink.getTitle(title)}"/>`;
+                });
+    
+                // Render result list
+                $('#plugin-fdk-resource-link-editor-results').html(items.join('\n'));
+            });
+
+            $('#plugin-fdk-resource-link-editor-search').on('input', () => {
+                const title = $('#plugin-fdk-resource-link-editor-search').val();
+                const { id, type } = $(`#plugin-fdk-resource-link-editor-results option[value='${title}']`).data();
+                if(id && type) {
+                    window.FdkResourceLink.selectedItem = { title, id, type };
+                }
+            })
+        },
+
+        getSearchBaseUrl: () => {
+            if (location.hostname.includes('staging')) {
+                return 'https://search.api.staging.fellesdatakatalog.digdir.no';
+            } else {
+                return 'https://search.api.fellesdatakatalog.digdir.no';
+            }
+        },
+
+        getFdkBaseUrl: () => {
+            if (location.hostname.includes('staging')) {
+                return 'https://staging.fellesdatakatalog.digdir.no';
+            } else {
+                return 'https://data.norge.no';
+            }
+        },
+
+        createLink: (textarea, selectionStart, selectionEnd) => {
+            if(!window.FdkResourceLink.selectedItem) {
+                return;
+            }
+
+            const { id, title, type } = window.FdkResourceLink.selectedItem;
+            const baseUrl = window.FdkResourceLink.getFdkBaseUrl();
+            var link;
+            switch (type) {
+                case 'DATASET':
+                    link = `[${title}](${baseUrl}/datasets/${id})`;
+                    break;
+                case 'DATA_SERVICE':
+                    link =  `[${title}](${baseUrl}/dataservices/${id})`;
+                    break;
+                case 'CONCEPT':
+                    link =  `[${title}](${baseUrl}/concepts/${id})`;
+                    break;
+                case 'INFORMATION_MODEL':
+                    link =  `[${title}](${baseUrl}/informationmodels/${id})`;
+                    break;
+                case 'EVENT':
+                    link =  `[${title}](${baseUrl}/public-services-and-events/${id})`;
+                    break;
+                case 'SERVICE':
+                    link =  `[${title}](${baseUrl}/public-services-and-events/${id})`;
+                    break;
```

Comment:
Dette skal linke til detaljsidene?
Det er bare søketreffene som har `public-services-and-events`, detaljsidene ligger under hhv `public-services`og `events`
```suggestion
                case 'EVENT':
                    link =  `[${title}](${baseUrl}/public-services/${id})`;
                    break;
                case 'SERVICE':
                    link =  `[${title}](${baseUrl}/events/${id})`;
                    break;
```